lib/vf/number.cpp

```diff
@@ -0,0 +1,50 @@
+/*
+ * Cppcheck - A tool for static C/C++ code analysis
+ * Copyright (C) 2007-2023 Cppcheck team.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "number.h"
+
+#include "token.h"
+#include "tokenlist.h"
+#include "valueflow.h"
+#include "vfvalue.h"
+
+#include "common.h"
+
+void ValueFlow::number(TokenList *tokenlist) {
```

Comment:
I dont think we should shorten the names like that as this could conflict with other functions in the future. I understand why we wouldn't want `ValueFlow::valueFlowNumber`, but we should have a clear naming scheme so they wont conflict.

Reply:
The other transformations have a naming scheme like `simplifySomething`. I am thinking something like `analyzeNumber` or `assignNumber`?

Reply:
That sounds good. Will adjust.

Reply:
done